---
title: "entrega1"
author: "Asier RÃ fales Vila"
date: "15/5/2019"
output: html_document 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(Biobase)

```

## summary 

## Title
Gene expression of distinct lung dendritic cell subsets.
## Organism
Mus musculus
## Experiment type
Expression profiling by array
## Summary
*Pulmonary dendritic cells are heterogenous cells comprise four distinct subsets including two conventional dendritic cell subsets, CD103+ and CD11bhiCD14lo cells, and two monocyte-derived dendritic cell subsets. Their functions in terms of migration and T cell activation are distinct, but genes regulating their features are to be determined.*
*We used microarrays to identify a select set of genes that are expressed in conventinal dendritic cells and in monocyte-derived dendriti cells.*
 	
## Overall design
Four distinct lung DC subsets were purified by flow cytometry-based sorting after inhalation of lipopolusaccharide and ovalbumin. Each subset has three replicates.
 	
## Contributor(s)	
Nakano H, Cook D

## Citation(s)	
Nakano H, Moran TP, Nakano K, Gerrish KE et al. Complement receptor C5aR1/CD88 and dipeptidyl peptidase-4/CD26 define distinct hematopoietic lineages of dendritic cells. J Immunol 2015 Apr 15;194(8):3808-19. PMID: 25769922


## Load Matrix into R environment 

```{r}
express <- read.delim("~/Documentos/uni/2n/3r_trimestre/omics techniques/2nd_part/Rafales_Asier_OmicsTechniques/Exercise_1/express.txt", row.names=1, comment.char="#")
```
  
Load targets 
```{r}
targets <- read.delim("~/Documentos/uni/2n/3r_trimestre/omics techniques/2nd_part/Rafales_Asier_OmicsTechniques/Exercise_1/targets.txt")
```





## Data description 

we have two files the first one, named express, contain a matrix with the information of rows (are diff gens or transcrits),  columns (are diff samples), so the numbers is the expression of the transcrip in each sample. 
as we can observe 



```{r}
summary(express)

```


Boxplot
```{r}
boxplot(express, col = c(2,2,2,3,3,3,4,4,4,5,5,5))
boxplot(log(express), col = c(2,2,2,3,3,3,4,4,4,5,5,5))
boxplot(express, col = c(2,2,2,3,3,3,4,4,4,5,5,5), outline = FALSE)
```



Hist

```{r}
#1
##GSM1582752
hist(express$GSM1582752)
hist(log(express$GSM1582752))
##GSM1582753
hist(express$GSM1582753)
hist(log(express$GSM1582753))
##GSM1582754
hist(express$GSM1582754)
hist(log(express$GSM1582754))

##GSM1582755
hist(express$GSM1582755)
hist(log(express$GSM1582755))

##GSM1582756
hist(express$GSM1582756)
hist(log(express$GSM1582756))

##GSM1582757
hist(express$GSM1582757)
hist(log(express$GSM1582757))

##GSM1582758
hist(express$GSM1582758)
hist(log(express$GSM1582758))

##GSM1582759
hist(express$GSM1582759)
hist(log(express$GSM1582759))

##GSM1582760
hist(log(express$GSM1582760))

##GSM1582761
hist(express$GSM1582761)
hist(log(express$GSM1582761))

##GSM1582762
hist(express$GSM1582762)
hist(log(express$GSM1582762))

##GSM1582763
hist(express$GSM1582763)
hist(log(express$GSM1582763))



```

Hist means

```{r}

row_mean <- rowMeans(log(express))
hist(row_mean)
## or
hist(express[,1],main="first row")
means <- apply(express,1,mean)
hist(log(means))
```





```{r}
if (!require(GEOquery)) {
  source("http://bioconductor.org/biocLite.R")
  biocLite("GEOquery")
  }
biocLite("GEOquery")
```









expression sets

My info
```{r}
## 
myInfo=list(myName="Asier Rafales", myLab="Bioinformatics Lab", 
          myContact="asier.rafales@alum.esci.upf.edu", myTitle="Practical Exercise on ExpressionSets")
show(myInfo)

```



```{r}
myGenes <- colnames(express)
head(myGenes)
```


```{r}
express2 <- data.matrix(express)
```



```{r}
head(express)
myEset <- ExpressionSet(express2)
class(myEset)
show(myEset)


```

```{r}
columnDesc <-  data.frame(labelDescription= c("Sample Names", "target","replicate","colors", "Age at disease onset(weeks)", "Sex of patient (male/female"))
columnDesc
myAnnotDF <- new("AnnotatedDataFrame", data=targets, varMetadata= columnDesc)
show(myAnnotDF)
```




```{r}
rownames(pData(myAnnotDF))<-pData(myAnnotDF)$Sample_name
myEset <- ExpressionSet(assayData=express2, phenoData=myAnnotDF)
show(myEset)
```


```{r}
myDesc <- new("MIAME", name= myInfo[["myName"]],
            lab= myInfo[["myLab"]],
            contact= myInfo[["myContact"]] ,
            title=myInfo[["myTitle"]])
print(myDesc)
```



```{r}
myEset <- ExpressionSet(assayData=express2, 
                        phenoData=myAnnotDF,
                        fetureNames =myGenes,
                        experimentData = myDesc)
show(myEset)

```

```{r}
dim(exprs(myEset))
class(phenoData(myEset))
class(pData(phenoData(myEset)))
head(pData(phenoData(myEset)))
head(pData(myEset))


```

```{r}
smallEset <- myEset[1:15,c(1:3,6:8)]
dim(exprs(smallEset))
dim(pData(smallEset))
head(pData(smallEset))
all(colnames(exprs(smallEset))==rownames(pData(smallEset)))

```


```{r}
youngEset <- myEset[,pData(myEset)$age<30]
dim(exprs(youngEset))
head(pData(youngEset))

```



```{r}
if (!require(GEOquery)) {
  BiocManager::install("GEOquery")
}
require(GEOquery)
gse <- getGEO("GSE64896")
class(gse)
names(gse)
gse[[1]]
esetFromGEO <- gse[[1]]

show(myEset)
```











