---
title: "entrega1"
author: "Asier RÃ fales Vila"
date: "15/5/2019"
output: 
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(Biobase)

```

# Series GSE64896

## Title:	Gene expression of distinct lung dendritic cell subsets
## Organism:	Mus musculus
## Experiment type:	Expression profiling by array
## Summary:
Pulmonary dendritic cells are heterogenous cells comprise four distinct subsets including two conventional dendritic cell subsets, CD103+ and CD11bhiCD14lo cells, and two monocyte-derived dendritic cell subsets. Their functions in terms of migration and T cell activation are distinct, but genes regulating their features are to be determined. We used microarrays to identify a select set of genes that are expressed in conventinal dendritic cells and in monocyte-derived dendriti cells.
 	
## Overall design:
Four distinct lung DC subsets were purified by flow cytometry-based sorting after inhalation of lipopolusaccharide and ovalbumin. Each subset has three replicates.
 	
## Contributor(s):	
*Nakano H, Cook D*
## Citation(s)	
*Nakano H, Moran TP, Nakano K, Gerrish KE et al. Complement receptor C5aR1/CD88 and dipeptidyl peptidase-4/CD26 define distinct hematopoietic lineages of dendritic cells. J Immunol 2015 Apr 15;194(8):3808-19. PMID: 25769922*

# Load Matrix into R environment 
Contains tha samples as columns and the transcripts as rows so we can see the expression of the transcripts in the different samples.
```{r}
express <- read.delim("~/Documentos/uni/2n/3r_trimestre/omics techniques/2nd_part/Rafales_Asier_OmicsTechniques/Exercise_1/express.txt", row.names=1, comment.char="#")
```
  
# Load targets 
Contains the sample_nam, target, replicate, gender	and age.
```{r}
targets <- read.delim("~/Documentos/uni/2n/3r_trimestre/omics techniques/2nd_part/Rafales_Asier_OmicsTechniques/Exercise_1/targets.txt")
```

# Data analysis 

## Sumary of matrix expressions.
```{r}
summary(express)

```

Here we can see that the means are similar in every sample, also that the min is a very low number and the max a very big number that difere so much of the first and third quartile so we can supose that there are some large outliers. 


## Boxplot
```{r}
boxplot(express, col = c(2,2,2,3,3,3,4,4,4,5,5,5))
```

In this first box plot we can't see anythig exept that the means are low compared with the outliers, and that there are very much ouliers ans someones are very large. This confirem our Asumption on the sumary of the matrix expressions. 

```{r}
boxplot(express, col = c(2,2,2,3,3,3,4,4,4,5,5,5), outline = FALSE)
```

Here we can se better tha plot, we remove the outliers so we confirm that the means and quartiles are similar.

```{r}
boxplot(log(express), col = c(2,2,2,3,3,3,4,4,4,5,5,5))
```

Here perform a log of the data to be able to see better the information. 


## Histograms 

```{r}
#1
##GSM1582752
hist(express$GSM1582752)
hist(log(express$GSM1582752))
##GSM1582753
hist(express$GSM1582753)
hist(log(express$GSM1582753))

```
With these 4 histograms of the two of the samples we can assume that the res will be similar.
The first histograms of each sample shows us that the majority of the expression is  between 0 and 5000
The second ones, I have used those to confirm what we observe in the log boxplots. 


# Hist means

```{r}

row_mean <- rowMeans(log(express))
hist(row_mean)
## or
hist(express[,1],main="first row")
means <- apply(express,1,mean)

```




# Install biocelite

```{r}
if (!require(GEOquery)) {
  source("http://bioconductor.org/biocLite.R")
  biocLite("GEOquery")
  }

```









# Expression sets
*Now we are going to make a dataset with all information.*

## My info
```{r}
## 
myInfo=list(myName="Asier Rafales", myLab="Bioinformatics Lab", 
          myContact="asier.rafales@alum.esci.upf.edu", myTitle="Practical Exercise on ExpressionSets")
show(myInfo)

```



```{r}
myGenes <- colnames(express)
head(myGenes)
```


```{r}
express2 <- data.matrix(express)
```



```{r}
head(express)
myEset <- ExpressionSet(express2)
class(myEset)
show(myEset)


```

```{r}
columnDesc <-  data.frame(labelDescription= c("Sample Names", "target","replicate","colors", "Age at disease onset(weeks)", "Sex of patient (male/female"))
columnDesc
myAnnotDF <- new("AnnotatedDataFrame", data=targets, varMetadata= columnDesc)
show(myAnnotDF)
```




```{r}
rownames(pData(myAnnotDF))<-pData(myAnnotDF)$Sample_name
myEset <- ExpressionSet(assayData=express2, phenoData=myAnnotDF)
show(myEset)
```


```{r}
myDesc <- new("MIAME", name= myInfo[["myName"]],
            lab= myInfo[["myLab"]],
            contact= myInfo[["myContact"]] ,
            title=myInfo[["myTitle"]])
print(myDesc)
```


## Finished set
```{r}
myEset <- ExpressionSet(assayData=express2, 
                        phenoData=myAnnotDF,
                        fetureNames =myGenes,
                        experimentData = myDesc)
show(myEset)

```


```{r}
dim(exprs(myEset))
class(phenoData(myEset))
class(pData(phenoData(myEset)))
head(pData(phenoData(myEset)))
head(pData(myEset))


```


## Small set.
```{r}
smallEset <- myEset[1:15,c(1:3,6:8)]
dim(exprs(smallEset))
dim(pData(smallEset))
head(pData(smallEset))
all(colnames(exprs(smallEset))==rownames(pData(smallEset)))

```

## Young set.

```{r}
youngEset <- myEset[,pData(myEset)$age<30]
dim(exprs(youngEset))
head(pData(youngEset))

```


## Second method to performa a dataset
```{r}
if (!require(GEOquery)) {
  BiocManager::install("GEOquery")
}
require(GEOquery)
gse <- getGEO("GSE64896")
class(gse)
names(gse)
gse[[1]]
esetFromGEO <- gse[[1]]

show(myEset)
```











